src/target-agents/dv360.ts

```diff
@@ -0,0 +1,266 @@
+   *
+   * @param {string} identifier
+   * @param {DV360_ENTITY_TYPE} type
+   * @param {boolean} evaluation
+   * @param {Parameters} params Additional parameters
+   */
+  process(
+    identifier: string,
+    type: DV360_ENTITY_TYPE,
+    evaluation: boolean,
+    params: Parameters
+  ) {
+    // Check for missing parameters
+    this.ensureRequiredParameters(params);
+
+    const auth = new Auth(params.serviceAccount ?? undefined);
+    this.authToken = auth.getAuthToken();
+
+    if (type === DV360_ENTITY_TYPE.LINE_ITEM) {
+      this.switchLIStatus_(params.advertiserId, identifier, evaluation);
+    } else if (type === DV360_ENTITY_TYPE.INSERTION_ORDER) {
+      this.switchIOStatus_(params.advertiserId, identifier, evaluation);
+    }
+  }
+
+  /**
+   * Check if supposed entity status matches its actual live status.
+   *
+   * @param {string} identifier
+   * @param {DV360_ENTITY_TYPE} type
+   * @param {boolean} evaluation
+   * @param {Parameters} params Additional parameters
+   * @returns {Array<string>}
+   */
+  validate(
+    identifier: string,
+    type: DV360_ENTITY_TYPE,
+    evaluation: boolean,
+    params: Parameters
+  ) {
+    // Check for missing parameters
+    this.ensureRequiredParameters(params);
+
+    const auth = new Auth(params.serviceAccount ?? undefined);
+    this.authToken = auth.getAuthToken();
+
+    let status;
+    const errors = [];
+
+    if (type === DV360_ENTITY_TYPE.LINE_ITEM) {
+      status = this.isLIActive_(params.advertiserId, identifier);
+    } else if (type === DV360_ENTITY_TYPE.INSERTION_ORDER) {
+      status = this.isIOActive_(params.advertiserId, identifier);
+    }
+
+    if (evaluation !== status) {
+      errors.push(
+        `Status for ${identifier} (${type}) should be ${evaluation} but is ${status}`
+      );
+    }
+
+    return errors;
+  }
+
+  /**
+   * Make an HTTP API request to the DV360 API.
+   *
+   * @param {string} url - API endpoint to be requested
+   * @param {string?} method - HTTP method, e.g. GET, PATCH, etc.
+   * @param {Object|undefined} payload - What should be updated
+   * @returns {JSON} Result of the operation
+   */
+  private fetchUrl_(url: string, method = 'get', payload?: Object | undefined) {
+    const headers = {
+      Authorization: `Bearer ${this.authToken}`,
+      Accept: '*/*',
+    };
+
+    return this.callApi(url, headers, undefined, payload, method);
+  }
+
+  /**
+   * Change DV360 entity status (Active/Paused) for the specified ID.
+   * See more: https://developers.google.com/display-video/api/reference/rest/v1/advertisers.lineItems
+   * See more: https://developers.google.com/display-video/api/reference/rest/v1/advertisers.insertionOrders
+   *
+   * @param {string} advertiserId DV360 Advertiser ID
+   * @param {string} entityId DV360 Line Item/Insertion Order ID
+   * @param {boolean} turnOn Activate on 'true', deactivate on 'false'
+   * @param {string} entity
+   */
+  private switchEntityStatus_(
```

Comment:
Proposal to reword: `setEntityStatusActive(advertiserId, entityId, active, entity)`
a) Currently the method does not switch the entity status.
b) Instructing behavior through a parameter (turnOn) seems unnecessary. Behavior should be part of the function name.
c) Not sure if "turnOn" is connotated. It's not turning me on ;)